test/app/components/task/details/output/task-output.component.spec.ts

```diff
@@ -0,0 +1,95 @@
+        fileServiceSpy = {
+            navigateTaskFile: jasmine.createSpy("navigateTaskFile").and.returnValue(fakeNavigator),
+        };
+
+        storageServiceSpy = {
+            navigateContainerBlobs: jasmine.createSpy("navigateContainerBlobs").and.returnValue(fakeNavigator),
+        };
+
+        TestBed.configureTestingModule({
+            imports: [RouterTestingModule],
+            declarations: [
+                TaskOutputsComponent, TestComponent,
+            ],
+            providers: [
+                { provide: FileService, useValue: fileServiceSpy },
+                { provide: StorageService, useValue: storageServiceSpy },
+            ],
+            schemas: [NO_ERRORS_SCHEMA],
+        });
+
+        fixture = TestBed.createComponent(TestComponent);
+        de = fixture.debugElement;
+        testComponent = fixture.componentInstance;
+        component = de.query(By.css("bl-task-outputs")).componentInstance;
+        fixture.detectChanges();
+    });
+
+    it("when task is active it should not start loading anything", () => {
+        testComponent.task = new Task({ id: "task-1", state: TaskState.active });
+        fixture.detectChanges();
+        expect(component.isTaskQueued).toBe(true);
+        expect(de.query(By.css("bl-file-explorer"))).toBeFalsy();
+        expect(de.query(By.css(".task-queued"))).not.toBeFalsy();
+        expect(fileServiceSpy.navigateTaskFile).not.toHaveBeenCalled();
+        expect(storageServiceSpy.navigateContainerBlobs).not.toHaveBeenCalled();
+    });
+
+    it("when task is preparing it should not start loading anything", () => {
+        testComponent.task = new Task({ id: "task-1", state: TaskState.preparing });
+        fixture.detectChanges();
+        expect(component.isTaskQueued).toBe(true);
+        expect(de.query(By.css("bl-file-explorer"))).toBeFalsy();
+        expect(de.query(By.css(".task-queued"))).not.toBeFalsy();
+        expect(fileServiceSpy.navigateTaskFile).not.toHaveBeenCalled();
+        expect(storageServiceSpy.navigateContainerBlobs).not.toHaveBeenCalled();
+    });
+
+    it("when task is running it should show the navigation", fakeAsync(() => {
+        testComponent.task = new Task({ id: "task-1", state: TaskState.running });
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+        expect(component.isTaskQueued).toBe(false);
+        expect(de.query(By.css("bl-file-explorer"))).not.toBeFalsy();
+        expect(de.query(By.css(".task-queued"))).toBeFalsy();
+
+        expect(fileServiceSpy.navigateTaskFile).toHaveBeenCalledOnce();
+        expect(fileServiceSpy.navigateTaskFile).toHaveBeenCalledWith("job-1", "task-1", jasmine.anything());
+        expect(storageServiceSpy.navigateContainerBlobs).toHaveBeenCalledOnce();
+        expect(storageServiceSpy.navigateContainerBlobs).toHaveBeenCalledWith("job-1", "task-1/", jasmine.anything());
+    }));
+});
```

Comment:
Have we make sure that the file size reloads from '-' to the actual size when the file becomes available. as per #523. if that's fixed then include this issue in the PR so it gets closed.